### PR TITLE
Add commands to load and list all archive files

### LINKED ISSUE #62: Command to view all archive files

### LINKED ISSUE #63: Command to load archive files

### PR DESCRIPTION
**Pull request by Johnwz123**
Closes #62 and Closes #63 

### List all archive files
Enter the command `listArchives` to list all archive filenames in the CLI.
If there are no archive files, it should return the message "No archive files found."

### Load archive file
Enter the command `loadArchive [FILENAME]` to load the respective archive file.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ListArchiveFilesCommand.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+
+/**
+ * Lists all archive files in the archive folder.
+ */
+public class ListArchiveFilesCommand extends Command {
+    private static final Logger logger = LogsCenter.getLogger(ListArchiveFilesCommand.class);
```
Comment (aldentantan) [inline src/main/java/seedu/address/logic/commands/ListArchiveFilesCommand.java:22]: The private variable should be declared after the public variables, following checkstyle.

File: src/main/java/seedu/address/logic/commands/LoadArchiveCommand.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import seedu.address.commons.exceptions.DataLoadingException;
+import seedu.address.model.Model;
+import seedu.address.model.ReadOnlyAddressBook;
+import seedu.address.storage.JsonAddressBookStorage;
+
+/**
+ * Loads an archive file and sets it as the current address book.
+ */
+public class LoadArchiveCommand extends Command {
+    private static final Logger logger = LogsCenter.getLogger(LoadArchiveCommand.class);
```
Comment (aldentantan) [inline src/main/java/seedu/address/logic/commands/LoadArchiveCommand.java:22]: Same comment as above, declare the private variable after all the public variables.

File: src/main/java/seedu/address/logic/commands/ListArchiveFilesCommand.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        Path archiveDir = Paths.get(source.getParent().toString(), "archive");
+        if (!Files.exists(archiveDir)) {
+            logger.info("No archive directory found.");
+            return new CommandResult(MESSAGE_NO_ARCHIVE);
+        }
+
+        List<String> archiveFiles = new ArrayList<>();
+
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(archiveDir, "*.json")) {
+            for (Path entry : stream) {
```
Comment (saajidshaik02) [inline src/main/java/seedu/address/logic/commands/ListArchiveFilesCommand.java:45]: Good use of streams!

File: src/main/java/seedu/address/logic/commands/LoadArchiveCommand.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    public LoadArchiveCommand(Filename archiveFilename) {
+        this.archiveFilename = archiveFilename;
+    }
+
+    @Override
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+
+        Path source = model.getAddressBookFilePath();
+        assert source != null : "Address book file path is null";
```
Comment (saajidshaik02) [inline src/main/java/seedu/address/logic/commands/LoadArchiveCommand.java:44]: Codecov / codecov/patch not covered, please do test for these cases!

**Review by saajidshaik02 — CHANGES_REQUESTED** [review]
Please see below

File: src/test/java/seedu/address/logic/commands/ArchiveCommandTest.java [hunk 5]
```diff
@@ -19,12 +19,21 @@
...
+        FileUtil.createIfMissing(new ModelManager().getAddressBookFilePath());
     }
 
     @AfterEach
     public void tearDown() throws IOException {
-        Files.deleteIfExists(new UserPrefs().getAddressBookFilePath());
+        Files.deleteIfExists(new ModelManager().getAddressBookFilePath());
+
+        // Delete all files in the archive directory
+        Files.walk(new ModelManager().getAddressBookFilePath().getParent())
```
Comment (kellywsq03) [inline src/test/java/seedu/address/logic/commands/ArchiveCommandTest.java:30]: Great use of method chaining. Keep up the good work.

File: src/test/java/seedu/address/logic/commands/ListArchiveFilesCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+    @Test
+    public void execute_withArchiveFiles_success() throws IOException {
+        Files.createFile(Path.of(archiveDir.toString(), "archive.json"));
+        String expectedMessage = ListArchiveFilesCommand.MESSAGE_SUCCESS + "\n" + "archive.json";
+        assertCommandSuccess(new ListArchiveFilesCommand(), modelStub, expectedMessage, modelStub);
+    }
+
+    @Test
+    @EnabledOnOs({OS.WINDOWS, OS.LINUX})
```
Comment (kellywsq03) [inline src/test/java/seedu/address/logic/commands/ListArchiveFilesCommandTest.java:65]: Will you be extended support to other platforms as well? What are the blockers?

File: src/main/java/seedu/address/logic/commands/LoadArchiveCommand.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+import seedu.address.storage.JsonAddressBookStorage;
+
+/**
+ * Loads an archive file and sets it as the current address book.
+ */
+public class LoadArchiveCommand extends Command {
+    public static final String COMMAND_WORD = "loadArchive";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Loads an archive file.\n"
+            + "Parameters: FILENAME\n"
+            + "Example: " + COMMAND_WORD + "addressbook-20241023_114324-example.json";
```
Comment (kellywsq03) [inline src/main/java/seedu/address/logic/commands/LoadArchiveCommand.java:25]: I apologize if this is a strange request as I'm unfamiliar with how the json files are named, but is it possible for the naming convention to be more user friendly? eg consistent dd/mm/yy format?

**Review by kellywsq03 — CHANGES_REQUESTED** [review]
Great work!

File: src/main/java/seedu/address/logic/commands/LoadArchiveCommand.java (hunk 7 above)
Comment (Johnwz123) [reply src/main/java/seedu/address/logic/commands/LoadArchiveCommand.java:25]: Noted. Have created an issue for it #79 
"20241023_114324" is the timestamp and "example" is the custom name FYI.

File: src/test/java/seedu/address/logic/commands/ArchiveCommandTest.java (hunk 5 above)
Comment (Johnwz123) [reply src/test/java/seedu/address/logic/commands/ArchiveCommandTest.java:30]: As discussed over Telegram.

**Review by kellywsq03 — APPROVED** [review]
LGTM :D

**Review by aldentantan — APPROVED** [review]

**Review by saajidshaik02 — APPROVED** [review]
LGTM!

**Pull request merged**